server/cmwell-dc/src/main/scala/cmwell/dc/stream/InfotonRetriever.scala

```diff
@@ -55,15 +53,32 @@ object InfotonRetriever extends LazyLogging {
   val initialRetrieveSingleStatus =
     RetrieveStateStatus(Settings.initialSingleRetrieveRetryCount, None)
 
-  case class ParsedNquad(path: String, nquad: ByteString, indexTime: Option[Long])
+  case class ExtraData(subject: String, indexTime: Option[Long], lastModifiedBy: Option[String]){
+    def shouldMerge (other: ExtraData): Option[ExtraData] = {
+      other match {
+        case ExtraData("", None, None) => None
+        case ExtraData(newSubject, None, None) if this.subject == "" => Some(this.copy(subject = newSubject))
+        case ExtraData(_, None, None) => None
+        case ExtraData(newSubject, Some(_), None) => Some(this.copy(subject = newSubject, indexTime = other.indexTime))
```

Comment:
The got subject should never be different from the current one - don't override it. Instead, throw an exception.

Reply:
It will be nicer to use the value got inside the Some (instead of _ and using `other`)